### DISCUSSION TITLE
Sha in Docker images name?

### DISCUSSION THREAD

**Discussion by danilopolani** [php/frankenphp#460] [Q&A] [answered]
Following the issue https://github.com/dunglas/frankenphp/issues/112 and the PR https://github.com/dunglas/frankenphp/pull/373 where a different naming was proposed, I was wondering why the Docker images are named with the sha in front of it and not just `php-8.2`. It would be definitely easier to read and find images called `dunglas/frankenphp:php-8.2` (or `dunglas/frankenphp:php8.2`) instead of `sha-f19c153-php8.2`.

Also, what's the difference between `sha-f19c153-php8.2` and `f19c153d06101b1ab7fb3b68f155c9f728d84a16-php8.2`?

**Comment by withinboredom** [comment, marked as answer]
Docker images with a `sha-*` prefix are specific commits of `main`, allowing someone to test a new version before it is released. You can use `dunglas/frankenphp:1.0.3-php8.3-{alpine|bookworm}` or just `dunglas/frankenphp:1.0.3-php8.3` for actual releases.

**Reply by withinboredom** [reply]
or even just `dunglas/frankenphp:1-php8.3` to pin to major version 1.

**Reply by danilopolani** [reply]
Oh thank you, I completely missed that, sorry! Do you know by any chance why the Docker Hub shows "last updated 9 days ago" for `dunglas/frankenphp:1.0-php8.2` and `dunglas/frankenphp:1.0.3-php8.2` even if the 1.0.3 was released two days ago? Maybe just a bug of Docker Hub?

**Reply by withinboredom** [reply]
Builds of stable artifacts run daily to pick up security fixes in the underlying PHP releases/os (https://github.com/dunglas/frankenphp/actions/runs/7443017166) just like regular PHP images (https://hub.docker.com/_/php/tags?page=1), however, we don't have a fancy trigger, so they just run on [a schedule](https://github.com/dunglas/frankenphp/actions/runs/7443017166/workflow#L19).

**Reply by withinboredom** [reply]
oh, also the last commit of main before 1.0.3 was probably 9 days ago, but since the image was tagged and was already uploaded (via the `sha-*` tag), it had already existed for awhile.

**Reply by danilopolani** [reply]
Got it, thank you again :)
